### DISCUSSION TITLE
onfocus event on inputs

### DISCUSSION THREAD

**Discussion by Rocky-25** [microsoft/fluentui-blazor#907] [Q&A]
I'm trying to trigger an event on focus of FluentSearch but think this would apply to all inputs. Firstly is it possible? @oninput works but @onfocus doesn't trigger when focusing on the control.

`  
<FluentSearch Id=@Id @bind-Value="Value" @onfocus=OnFocusEvent @oninput="OnInput">
  </FluentSearch>
`

**Comment by vnbaaij** [comment]
Hi,

We are currently not supporting `@onfocus` as a parameter on any of the input components. We can think about adding them if this question comes up more (you're the first).

**Comment by Hwiet** [comment]
I was looking for this too for FluentTextField. `@onfocus` and `@onblur` handlers on FluentTextField were not firing.

I found [this workaround](https://github.com/dotnet/aspnetcore/issues/39561#issuecomment-1014561262) to use `@onfocusin` instead of `@onfocus` and `@onfocusout` instead of `@onblur`, which works for my needs.
